### PR TITLE
Implement the `eth_getCode` JSON-RPC endpoint

### PR DESCRIPTION
**Pull request by m-Peter**
Closes: https://github.com/onflow/flow-evm-gateway/issues/109

## Description

This endpoint can be used to fetch the deployed code at a given address.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: services/requester/requester.go [hunk 1]
```diff
@@ -380,6 +387,50 @@ func (e *EVM) EstimateGas(ctx context.Context, data []byte) (uint64, error) {
...
+
+	hexEncodedCode := value.ToGoValue().(string)
+	code, err := hex.DecodeString(hexEncodedCode)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert string to byte code: %w", err)
+	}
+
+	e.logger.Info().
+		Str("address", address.Hex()).
+		Str("result", hex.EncodeToString(code)).
```
Comment (sideninja) [inline services/requester/requester.go:428]: I would drop the result, it will be noisy on the logs.

File: services/requester/requester.go [hunk 2]
```diff
@@ -380,6 +387,50 @@ func (e *EVM) EstimateGas(ctx context.Context, data []byte) (uint64, error) {
...
+	ctx context.Context,
+	address common.Address,
+	height uint64,
+) ([]byte, error) {
+	e.logger.Debug().
+		Str("addess", address.Hex()).
+		Str("height", fmt.Sprintf("%d", height)).
+		Msg("get code")
+
+	hexEncodedAddress, err := cadence.NewString(
```
Comment (sideninja) [inline services/requester/requester.go:400]: I think this is getting repetitive, lets just add a helper function at the bottom of this file something like: `addressToCadenceString`

File: services/requester/requester.go [hunk 3]
```diff
@@ -380,6 +387,50 @@ func (e *EVM) EstimateGas(ctx context.Context, data []byte) (uint64, error) {
...
+
+	if _, ok := value.(cadence.String); !ok {
+		e.logger.Panic().Msg(fmt.Sprintf("failed to convert value to hex-encoded string: %v", value))
+	}
+
+	hexEncodedCode := value.ToGoValue().(string)
+	code, err := hex.DecodeString(hexEncodedCode)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert string to byte code: %w", err)
+	}
```
Comment (sideninja) [inline services/requester/requester.go:424]: maybe same for decode, a helper function that is something like: `cadenceStringToBytes()`

**Review by sideninja — APPROVED** [review]
Nice, left some comments.

File: services/requester/requester.go (hunk 2 above)
Comment (m-Peter) [reply services/requester/requester.go]: Updated in https://github.com/onflow/flow-evm-gateway/pull/130/commits/3929b93a2a772ee336bef963ed5f2605da62447f

File: services/requester/requester.go (hunk 3 above)
Comment (m-Peter) [reply services/requester/requester.go]: Updated in https://github.com/onflow/flow-evm-gateway/pull/130/commits/3929b93a2a772ee336bef963ed5f2605da62447f

**Pull request merged**